devlib/utils/types.py

```diff
@@ -68,6 +68,15 @@ def numeric(value):
     """
     if isinstance(value, int):
         return value
+
+    if isinstance(value, basestring):
+        value = value.strip()
+        if value.endswith('%'):
+            try:
+                return float(value.rstrip('%')) / 100
```

Comment:
The `int` / `float` thing below can be replaced by a single call to `ast.literal_eval()` which will return an `int` or a `float` when appropriate.